vm/src/exceptions.rs

```diff
@@ -1621,3 +1622,9 @@ pub(super) mod types {
         "Base class for warnings about encodings."
     }
 }
+
+impl ToPyException for ReprOverflowError {
+    fn to_pyexception(&self, vm: &VirtualMachine) -> PyBaseExceptionRef {
+        vm.new_value_error(String::from("This object has overflow repr"))
```

Comment:
use an `Overflow` error instead here (unless Python explicitly raises this ValueError, I haven't checked)